Lib/test/test_netrc.py

```diff
@@ -123,6 +124,33 @@ def test_security(self):
             os.chmod(fn, 0o622)
             self.assertRaises(netrc.NetrcParseError, netrc.netrc)
 
+    def test_file_not_found_in_home(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        with support.EnvironmentVarGuard() as environ:
+            environ.set('HOME', d)
+            self.assertRaises(FileNotFoundError, netrc.netrc)
+
+    def test_file_not_found_explicit(self):
+        self.assertRaises(FileNotFoundError, netrc.netrc,
+                          file="unlikely_netrc")
+
+    def test_home_not_set(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        fake_netrc_path = os.path.join(d, 'netrc')
+        with open(fake_netrc_path, 'w') as f:
+            f.write("machine foo.domain.com login bar password pass")
+        os.chmod(fake_netrc_path, 0o600)
+        with support.EnvironmentVarGuard() as environ:
+            environ.unset('HOME')
+            with mock.patch('os.path.expanduser') as mock_expanduser:
```

Comment:
Mocking `os.path.expanduser` is not needed here in my opinion. If you can change the test to

```py
def test_home_not_set(self):
    with support.EnvironmentVarGuard() as environ:
        environ.unset('HOME')
        self.assertRaises(FileNotFoundError, netrc.netrc)
```

then we are good to go! :)